pulpcore/cli/ansible/distribution.py

```diff
@@ -0,0 +1,133 @@
+
+
+@distribution.command()
+@click.option("--name", required=True)
+@click.option("--base-path", required=True)
+@click.option("--repository", help="name of repository")
+@click.option("--version", help="version of repository, leave blank for always latest")
+@click.option("--content-guard")
+@pass_entity_context
+@pass_pulp_context
+def create(
+    pulp_ctx: PulpContext,
+    distribution_ctx: PulpAnsibleDistributionContext,
+    name: str,
+    base_path: str,
+    repository: Optional[str],
+    version: Optional[str],
+    content_guard: Optional[str],
+) -> None:
+    body: Dict[str, Any] = {"name": name, "base_path": base_path}
+    repo: Dict[str, Any] = PulpAnsibleRepositoryContext(pulp_ctx).find(name=repository)
+    if version:
+        body["repository_version"] = f'{repo["versions_href"]}{version}/'
+    elif repository:
+        body["repository"] = repo["pulp_href"]
+    if content_guard:
+        body["content_guard"] = content_guard
+    result = distribution_ctx.create(body=body)
+    distribution = distribution_ctx.show(result["created_resources"][0])
+    pulp_ctx.output_result(distribution)
+
+
+@distribution.command()
+@click.option("--href", required=True)
+@click.option("--name")
+@click.option("--base-path")
+@click.option("--repository", type=str, default=None)
+@click.option("--version", type=str, default=None)
+@click.option("--content-guard", type=str, default=None)
+@pass_entity_context
+@pass_pulp_context
+def update(
+    pulp_ctx: PulpContext,
+    distribution_ctx: PulpAnsibleDistributionContext,
+    href: str,
+    name: str,
+    base_path: str,
+    repository: Optional[str],
+    version: Optional[str],
+    content_guard: Optional[str],
+) -> None:
+    """
+    To remove repository or repository_version set --repository to ""
+    """
+    dist_body = distribution_ctx.show(href)
+    body = dict()
+    body["name"] = name or dist_body["name"]  # This is full update so these are required
+    body["base_path"] = base_path or dist_body["base_path"]
+    if repository is not None:
+        if repository == "":
+            # unset repository or repository version
+            if dist_body["repository"]:
+                body["repository"] = ""
+            elif dist_body["repository_version"]:
+                body["repository_version"] = ""
+        else:
+            repo = PulpAnsibleRepositoryContext(pulp_ctx).find(name=repository)
+            if version:
+                if dist_body["repository"]:
+                    unset_repository(pulp_ctx, distribution_ctx, href, "repository")
+                body["repository_version"] = f'{repo["versions_href"]}{version}/'
+            else:
+                if dist_body["repository_version"]:
+                    unset_repository(pulp_ctx, distribution_ctx, href, "repository_version")
+                body["repository"] = repo["pulp_href"]
+    if content_guard is not None:
+        body["content_guard"] = content_guard
```

Comment:
This is not referring to the content guard by it's name.
You'd need to add a corresponding context class and do the lookup (with c_g_ctx.find()) here.